community/index/src/main/java/org/neo4j/index/gbptree/GBPTree.java

```diff
@@ -447,25 +441,31 @@ public void merge( KEY key, VALUE value, ValueMerger<VALUE> valueMerger ) throws
         {
             goToRoot( cursor );
 
-            SplitResult<KEY> split = treeLogic.insert( cursor, key, value, valueMerger, options,
+            treeLogic.insert( cursor, structurePropagation, key, value, valueMerger, options,
```

Comment:
`structurePropagation` is storage of split result that particular insert/remove cause for a tree and depending from results of that we will do some post steps, it also means that if we will forget to clear it or will fail to do that that can potentially influence future step. Can we have block with reset of that context object to clean initial state in finally section or before using it?